uml/sequence_diagrams/ClockView.txt

```diff
@@ -0,0 +1,46 @@
+@startuml
+participant DeactivatedWatch 
+participant DisplaySelection 
+participant ClockView 
+participant BlueTooth
+participant Battery
+participant DisplayError 
+
+
+opt Activation
+  DeactivatedWatch -> ClockView:  Activate
+end
+opt Clock
+  DisplaySelection --> ClockView : GoTo Clock
+end
+loop event
+alt event == askTimeEvent
+ClockView --> BlueTooth: Ask time
+Activate BlueTooth
+BlueTooth --> ClockView: Time
+Deactivate BlueTooth
+opt Time == TimeError
+  ClockView --> DisplayError : SendError(TimeError)
+  Activate DisplayError
+  DisplayError --> ClockView : DefaultTime (00:00)
+  Deactivate DisplayError
+end
+ClockView --> ClockView: setDisplayTime(Time)
+else event = AskBatteryEvent
+ClockView --> Battery: AskBattery
+Activate Battery
+Battery --> ClockView: Battery

```

Comment:
Might be best to have `BatterySoC` as the data transmitted. 

Also, after you receive this, might be worth doing the following:

`Battery --> Battery: setDisplayCharge(BatterySoC)`

(as you have done with the received time)

Note: I have updated my state diagram to refer to "Battery Error" - same terminology as you use.

Reply:
Fixed. Also in power sequence.